src/controllers/scripting/legacy/scriptconnection.cpp

```diff
@@ -5,19 +5,32 @@
 
 void ScriptConnection::executeCallback(double value) const {
     Trace executeCallbackTrace("JS %1 callback", key.item);
-    const auto args = QJSValueList{
-            value,
-            key.group,
-            key.item,
-    };
-    QJSValue func = callback; // copy function because QJSValue::call is not const
-    QJSValue result = func.call(args);
-    if (result.isError()) {
-        if (controllerEngine != nullptr) {
-            controllerEngine->showScriptExceptionDialog(result);
+    if (!m_preventRecursiveCalls) {
+        if (!m_callbackExecuting) {
+            m_callbackExecuting = true;
```

Comment:
m_callbackExecuting initialization is missing in the constructor of ScriptConnection.

Reply:
I have implemented both m_preventRecursiveCalls and m_callbackExecuting now , I will send you a clean PR which will be understandable.

Reply:
Thank you.